sui_core/src/checkpoints/mod.rs

```diff
@@ -0,0 +1,1095 @@
+/// from real consensus.
+pub trait ConsensusSender: Send + Sync + 'static {
+    // Send an item to the consensus
+    fn send_to_consensus(&self, fragment: CheckpointFragment) -> Result<(), SuiError>;
+}
+
+#[derive(Debug)]
+pub enum FragmentInternalError {
+    Error(SuiError),
+    Retry(Box<CheckpointFragment>),
+}
+
+pub struct CheckpointStore {
+    // Fixed size, static, identity of the authority
+    /// The name of this authority.
+    pub name: AuthorityName,
+    /// Committee of this Sui instance.
+    pub committee: Committee,
+    /// The signature key of the authority.
+    pub secret: StableSyncAuthoritySigner,
+
+    /// The list of all transactions that are checkpointed mapping to the checkpoint
+    /// sequence number they were assigned to.
+    pub transactions_to_checkpoint:
+        DBMap<TransactionDigest, (CheckpointSequenceNumber, TxSequenceNumber)>,
+
+    /// The mapping from checkpoint to transactions contained within the checkpoint.
+    /// The second part of the key is the local sequence number if the transaction was
+    /// processed or Max(u64) / 2 + offset if not. It allows the authority to store and serve
+    /// checkpoints in a causal order that can be processed in order. (Note the set
+    /// of transactions in the checkpoint is global but not the order.)
+    pub checkpoint_contents: DBMap<(CheckpointSequenceNumber, TxSequenceNumber), TransactionDigest>,
+
+    /// The set of pending transactions that were included in the last checkpoint
+    /// but that this authority has not yet processed.
+    pub unprocessed_transactions: DBMap<TransactionDigest, CheckpointSequenceNumber>,
+
+    /// The set of transactions this authority has processed but have not yet been
+    /// included in a checkpoint, and their sequence number in the local sequence
+    /// of this authority.
+    pub extra_transactions: DBMap<TransactionDigest, TxSequenceNumber>,
+
+    /// The list of checkpoint, along with their authentication information
+    pub checkpoints: DBMap<CheckpointSequenceNumber, AuthenticatedCheckpoint>,
+
+    // --- Logic related to fragments on the way to making checkpoints
+
+    // A list of own fragments indexed by the other node that the fragment connects
+    // to. These are used for the local node to potentially reconstruct the full
+    // transaction set.
+    pub local_fragments: DBMap<AuthorityName, CheckpointFragment>,
+
+    /// Store the fragments received in order, the counter is purely internal,
+    /// to allow us to provide a list in order they were received. We only store
+    /// the fragments that are relevant to the next checkpoints. Past checkpoints
+    /// already contain all relevant information from previous checkpoints.
+    pub fragments: DBMap<u64, CheckpointFragment>,
+
+    /// The local sequence at which the proposal for the next checkpoint is created
+    /// This is a sequence number containing all unprocessed trasnactions lower than
```

Comment:
```suggestion
    /// This is a sequence number containing all unprocessed transactions lower than
```

Reply:
I did a search and found 6 different places where I misspell it this way. All fixed!